Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.XR;
+using System;
+using UnityEngine.XR.WSA.Input;
+
+#if HP_CONTROLLER_ENABLED
+using Microsoft.MixedReality.Input;
+using MotionControllerHandedness = Microsoft.MixedReality.Input.Handedness;
+using Handedness = Microsoft.MixedReality.Toolkit.Utilities.Handedness;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input
+{
+#if HP_CONTROLLER_ENABLED
```

Comment:
Since this class has the `else` case that calls into `base.UpdateController(interactionSourceState);` when the `MotionControllerState` isn't found, could this class be wrapped in this `#if` more specifically (specifically around the lines that refer to types defined there) and use the above `else` case otherwise? Same question for the XR SDK version

Reply:
Wrapped around the if else logic in a version define, so the extra check isn't done if HP_CONTROLLER_ENABLED isn't defined. The check still exists otherwise, since it's possible for the MotionControllerState to not be immediately available when instantiating a HPMotionController